File: 2109-adding-spaces-to-a-string/2109-adding-spaces-to-a-string.js
/**
 * @param {string} s
 * @param {number[]} spaces
 * @return {string}
 */
var addSpaces = function(s, spaces) {
    let ans ="";
    let j=0;
    for(let i=0;i<spaces.length;i++)
    {
        while(j<s.length && j<spaces[i])
        {
            ans+=s[j];
            j++;
        }
        if(j==spaces[i])
        {
            ans+=" ";
        }
    }
    while(j<s.length)
    {
        ans+=s[j];
        j++;
    }
    return ans;
};

File: 2109-adding-spaces-to-a-string/2109-adding-spaces-to-a-string.cpp
class Solution {
public:
    string addSpaces(string s, vector<int>& spaces) {
        int j=0;
        string ans="";
    for(auto i:spaces)
    {
         ans+=s.substr(j,i-j)+" " ; 
         j=i;
    }
    ans+=s.substr(j);
     return ans;
    }
};